### DISCUSSION TITLE
One build script for all

### DISCUSSION THREAD

**Discussion by omdxp** [dzcode-io/kuliya#102] [Ideas]
I was observing lately with our implementations of Kuliya for each language, and they almost share the same build step which lead me to this idea:

Why not having one build script that is language agnostic (changes would happen to this script whenever there is a new language implementation). It can be a CLI script that can be run like this:

```sh
./build --lang rust -o ./rust/src/data.rs
```

And as you may imagine, this can be applied to all languages implementations that will eliminate replicating the build script per each language.

Any thoughts?

**Comment by ZibanPirate** [comment]
the idea of this repo is to solve the same problem with different languages, this includes the build step (massaging the `_data` folder to some other format), so the duplication here is justified in my opinion, and this way, you don't need another tool to run the build step, you only need the tools for that specific language you want to work.

**Reply by omdxp** [reply]
I see your point
